scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+2.  It can be challenging to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process can be complex, as it may require patch release cuts, changes to different branches, and integration between logic that is managed in the master branch, executing other releases binaries.
+3.  The latest offline bundles does not support post `4.5` releases.  Bundle embedding for each release increases the cost to store and consume offline bundles.  The `roxctl` interface and documentation today needs to be clearer on these limitations.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Source Ref   | Description                                                     |
+|--------|--------------|-----------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema              |
+| v1     | tags/4.5.0   | This stream tracks the schema used by the last StackRox release |
+
+On the subsequent releases (for example, `4.6.0`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release branch (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.6.0   |
+
+2.  The schema has changed, so v1 stays at the current git ref (`tags/4.5.0`), and a new version, v2, is created to track the next release (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.5.0   |
+   | v2     | tags/4.6.0   |
+
+This process could be either manual or automated.
+
+### Building versioned bundles
+
+#### Online mode
+
+The GitHub workflow will have a map with "bundle version" as keys and "source reference" as values, similar to Table 1 above.  This map can be stored statically in the matrix definition or as a file in the source repository.
+
+A source reference can be one of the following:
+
+1.  A branch reference, `heads/<branch-name>`, example: `heads/release-4.6`
+2.  A tag reference, `tags/<tag-name>`, example: `tags/4.6.0`.
+3.  A StackRox release, `release/<stackrox-release>`, example: `release/4.6.0`
+
+Both (1.) and (2.) will determine a specific commit to use to build the vulnerability exporter.
+
+(3.) will select a desired release build based on a StackRox target release.  The workflow will find the best matching Git tag for the specified `<stackrox-release>`, by checking if the version exists as a tag or matching it against release candidate (`-rc`) or patch (`.x`) tags.
```

Comment:
I understand the rc tags, but I'm confused by the patch tags. Wouldn't we just use `tags/4.6.1` for a patch release?

Reply:
We can, but during the release, the tag is not cut until the release is ready. Using `<stackrox-release>` allow us to start building bundles for that release based on the release candidates upfront, without having to update this mapping manually.

Reply:
Can you give an example for when the `.x` patch tag would be used? I'm still having a hard time understanding this part completely

Reply:
It takes precedence over the `-rc` tags. An typically it's the parent of the release branch commit path/lineage. Although, for 4.5, it was tagged `4.6.x`:

![image](https://github.com/user-attachments/assets/ac09c86c-4f94-4c93-a946-0d45dfa72008)

Reply:
ah ok, so `x` is not a variable in this case, but literally the letter `x`. Makes sense

Though, what you mention may be something to think about a bit more: I do recall discussion (and I guess an active change) to switch our development versioning scheme. Once a release branch cut, like 4.4, then all subsequent versions were tagged with something like 4.4.x-<whatever>. However, starting with 4.5, I believe it was decided to do `minor + 1`, so it's currently `4.6.x-<whatever>` instead of `4.5.x-<whatever>`. I believe the idea was to make it clear you are now developing for `4.6` instead of for `4.5`.

Given this, for `4.5+`, this may need to be adjusted to `4.<minor + 1>.x-<whatever>`. Is that worth updating here, or is that more of an implementation detail?

Reply:
Interesting. I need to follow-up on this new versioning schema. And I think I will leave it as an implementation detail, best to make the ADR last longer instead of us having outdated docs, or having to catch up with new versioning/tag systems.

Reply:
I will resolve this, but took https://issues.redhat.com/browse/ROX-26053